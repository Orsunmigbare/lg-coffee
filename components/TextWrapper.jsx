import React from 'react';
import PropTypes from 'prop-types';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

const TextWrapper = ({ children }) => (
  <Row>
    <Col
      lg={{ span: 6, offset: 3 }}
      md={{ span: 8, offset: 2 }}
      sm={{ span: 10, offset: 1 }}
      xs={12}
    >
      {children}
    </Col>
  </Row>
);

TextWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TextWrapper;
