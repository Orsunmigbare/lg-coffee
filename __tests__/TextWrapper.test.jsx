import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TextWrapper from '../components/TextWrapper';

it('TextWrapper component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextWrapper><h1>Hello World</h1></TextWrapper>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('TextWrapper component renders unchanged', () => {
  const tree = renderer.create(<TextWrapper><h1>Hello World</h1></TextWrapper>).toJSON();
  expect(tree).toMatchSnapshot();
});
