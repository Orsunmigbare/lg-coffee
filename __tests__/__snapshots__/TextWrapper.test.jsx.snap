// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWrapper component renders unchanged 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6 col-md-8 col-sm-10 col-12 offset-lg-3 offset-md-2 offset-sm-1"
  >
    <h1>
      Hello World
    </h1>
  </div>
</div>
`;
